march_gait_selection/test/nosetests/test_gait_selection.py

```diff
@@ -1,78 +1,89 @@
-
-    def test_right_close_walk_small(self):
-        # check if subgait name exists
-        is_right_close_subgait = False if gait_selection[walk_small][right_close] is None else True
-        self.assertTrue(is_right_close_subgait, msg='{sg} subgait could not be found in gait {gn}'
-                        .format(sg=right_close, gn=walk_small))
-
-    def test_right_close_walk_medium(self):
-        # check if subgait name exists
-        is_right_close_subgait = False if gait_selection[walk_medium][right_close] is None else True
-        self.assertTrue(is_right_close_subgait, msg='{sg} subgait could not be found in gait {gn}'
-                        .format(sg=right_close, gn=walk_medium))
+
+    @classmethod
+    def setUpClass(cls):
+        cls._gait_selection = GaitSelection(VALID_PACKAGE, VALID_DIRECTORY)
+
+    def setUp(self):
+        self.gait_selection = deepcopy(self._gait_selection)
+        self.gait_version_map = deepcopy(self._gait_selection.gait_version_map)
+
+    # __init__ tests
+    def test_init_with_wrong_package(self):
+        with self.assertRaises(PackageNotFoundError):
+            GaitSelection('wrong', VALID_DIRECTORY)
+
+    def test_init_with_wrong_directory(self):
+        with self.assertRaises(FileNotFoundError):
+            GaitSelection(VALID_PACKAGE, 'wrong')
+
+    # gait version map setter tests
+    def test_set_gait_version_map_with_wrong_type(self):
+        with self.assertRaises(TypeError):
+            self.gait_selection.gait_version_map = 'wrong'
+
+    def test_set_gait_version_map_with_empty_version_map(self):
+        self.gait_version_map = {}
+        with self.assertRaises(GaitError):
+            self.gait_selection.gait_version_map = self.gait_version_map
+
+    def test_set_gait_version_map_with_non_existing_gait_name(self):
+        self.gait_version_map['wrong'] = self.gait_version_map['walk']
+        with self.assertRaises(GaitError):
+            self.gait_selection.gait_version_map = self.gait_version_map
+
+    def test_set_gait_version_map_with_non_existing_subgait_name(self):
+        self.gait_version_map['walk']['wrong'] = self.gait_version_map['walk']['right_open']
+        with self.assertRaises(GaitError):
+            self.gait_selection.gait_version_map = self.gait_version_map
+
+    def test_set_gait_version_map_with_wrong_subgait_version(self):
+        self.gait_version_map['walk']['right_open'] = 'wrong'
+        with self.assertRaises(GaitError):
+            self.gait_selection.gait_version_map = self.gait_version_map
+
+    # load gaits tests
+    def test_types_in_loaded_gaits(self):
+        for gait in self.gait_selection.loaded_gaits:
+            self.assertIsInstance(gait, (Gait, BalanceGait))
+
+    # scan directory tests
+    def test_scan_directory_top_level_length(self):
+        directory = self.gait_selection.scan_directory()
+        self.assertTrue(len(['stairs_up', 'walk', 'walk_medium', 'walk_small']), len(directory.keys()))
```

Comment:
```suggestion
        self.assertEqual(len(['stairs_up', 'walk', 'walk_medium', 'walk_small']), len(directory.keys()))
```
What do you want to test here? why not test if len(directory.keys()) == 4 or
directory.keys() == ['stairs_up', 'walk', 'walk_medium', 'walk_small']

Reply:
If all the maps included in the test resources are parsed properly!